tests/core/disabled_non_linear_solver.cc

```diff
@@ -0,0 +1,89 @@
+// SPDX-FileCopyrightText: Copyright (c) 2025 The Lethe Authors
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception OR LGPL-2.1-or-later
+
+/**
+ * @brief The TestClass tests the non-linear solvers using a simple system of two
+ * equations, only one of which is non-linear
+ */
+
+// Lethe
+#include <core/parameters.h>
+
+// Tests
+#include <../tests/core/linear_test_system_01.h>
+
+#include <../tests/tests.h>
+
+void
+test()
+{
+  Parameters::NonLinearSolver params{
+    .verbosity = Parameters::Verbosity::verbose,
+    .solver    = Parameters::NonLinearSolver::SolverType::disabled,
+    .kinsol_strategy =
+      Parameters::NonLinearSolver::KinsolStrategy::normal_newton, // not used in
+                                                                  // this case
+    .tolerance                    = 1e-8,   // not used in this case
+    .max_iterations               = 10,     // not used in this case
+    .display_precision            = 4,      // not used in this case
+    .force_rhs_calculation        = false,  // not used in this case
+    .matrix_tolerance             = 0.1,    // not used in this case
+    .step_tolerance               = 0.99,   // not used in this case
+    .reuse_matrix                 = false,  // not used in this case
+    .reuse_preconditioner         = false,  // not used in this case
+    .abort_at_convergence_failure = false}; // not used in this case
+
+  deallog << "Creating solver" << std::endl;
+
+  // Create an instantiation of the Test Class
+  std::unique_ptr<LinearProblemTestClass> solver =
+    std::make_unique<LinearProblemTestClass>(params);
+
+
+  deallog << "Solving linear system " << std::endl;
+  // We use solve_non_linear_system of equation even if the system is linear
+  // because it inherits from the NonLinearSolver base class.
+  solver->solve_non_linear_system(true);
```

Comment:
I guess here in the future a better name could be solve_system instead of solve_non_linear system.
In a way the non-linear equations solver are solvers for system of equations :). Maybe a follow-up PR.